components/sections/ExportTranslations.tsx

```diff
@@ -64,44 +66,61 @@ const Separator = () => (
   </div>
 );
 
-const ExportTranslations = ({ translations, categoryName, triggerLabel }: ExportTranslationsProps) => {
+const ExportTranslations = ({ dictionary, triggerLabel, category }: ExportTranslationsProps) => {
   const [isModalOpen, setIsModalOpen] = useState(false);
   const [translationSeparator, setTranslationSeparator] = useState(TRANSLATION_SEPARATORS[0].value);
   const [customTranslationSeparator, setCustomTranslationSeparator] = useState(' - ');
   const [phraseSeparator, setPhraseSeparator] = useState(PHRASE_SEPARATORS[0].value);
   const [customPhraseSeparator, setCustomPhraseSeparator] = useState('\\n\\n');
   const [includeTranscriptions, setIncludeTranscriptions] = useState(false);
   const { currentLanguage, otherLanguage } = useLanguage();
+  const { t } = useTranslation();
 
   const translSep = translationSeparator === TRANS_SEP_CUSTOM ? customTranslationSeparator : translationSeparator;
   const phraseSep = phraseSeparator === PHRASE_SEP_CUSTOM ? customPhraseSeparator : phraseSeparator;
-  const phrases = translations
-    .map(
-      (translation) =>
-        translation.getTranslation(currentLanguage) +
-        (includeTranscriptions ? ` [${translation.getTranscription(currentLanguage)}]` : '') +
-        translSep +
-        translation.getTranslation(otherLanguage) +
-        (includeTranscriptions ? ` [${translation.getTranscription(otherLanguage)}]` : '') +
-        phraseSep
-    )
-    .map((translation) => unescapeTabsAndNewlines(translation));
+  const constructPhrase = (phraseId: PhraseDataObject['id']) => {
+    const phrase = getPhraseById(dictionary, phraseId);
+    const output =
+      phrase.getTranslation(currentLanguage) +
+      (includeTranscriptions ? ` [${phrase.getTranscription(currentLanguage)}]` : '') +
+      translSep +
+      phrase.getTranslation(otherLanguage) +
+      (includeTranscriptions ? ` [${phrase.getTranscription(otherLanguage)}]` : '') +
+      phraseSep;
+
+    return unescapeTabsAndNewlines(output);
+  };
+
+  const constructCategoryHeader = (category: CategoryDataObject) =>
+    currentLanguage === 'uk'
+      ? '[' + category.name.source + ']' + translSep + '[' + category.name.main + ']' + '\n'
+      : '[' + category.name.main + ']' + translSep + '[' + category.name.source + ']' + '\n';
+
+  const source = category !== undefined ? dictionary.categories.filter((c) => c.id === category.id) : dictionary.categories;
+
+  const categories = source.map((category, index, arr) => [
```

Comment:
I decided to store formatted phrases within categories in array, then I have array for each category which contains header, phrases --> 2D array, and category separator. This approach lets me easily extract phrases from category which is handy for preview.